# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 695d708318a5
# Full Hash: 695d708318a576da278e5c035f77503274e49b7b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 09:46:07
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 5-3: Make `HTMLEditor::HandleCSSIndentAtSelectionInternal` stop touching `Selection` directly and setting `mNewBlockElement` r=m_kato
#   
#   It's called only by `HTMLEditor::HandleCSSIndentAtSelection` which is called
#   only by `HTMLEditor::HandleIndentAtSelection`.  They don't touch `Selection`
#   after calling it.  Therefore, we can make it adjust collapsing selection point
# ==============================================================================

diff -r 80ed14c716a2 -r 695d708318a5 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Thu Aug 04 03:00:31 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Thu Aug 04 03:16:28 2022 +0000
@@ -2319,16 +2319,13 @@
                     const Element& aEditingHost);
 
   /**
-   * HandleCSSIndentAtSelectionInternal() indents around Selection with CSS.
-   * This method creates AutoSelectionRestorer.  Therefore, each caller
-   * need to check if the editor is still available even if this returns
-   * NS_OK.
-   * NOTE: Use HandleCSSIndentAtSelection() instead.
+   * HandleCSSIndentAroundRanges() indents around aRanges with CSS.
    *
+   * @param aRanges             The ranges where the content should be indented.
    * @param aEditingHost        The editing host.
    */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
-  HandleCSSIndentAtSelectionInternal(const Element& aEditingHost);
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult HandleCSSIndentAroundRanges(
+      AutoRangeArray& aRanges, const Element& aEditingHost);
 
   /**
    * HandleHTMLIndentAtSelectionInternal() indents around Selection with HTML.
@@ -2343,17 +2340,6 @@
   HandleHTMLIndentAtSelectionInternal(const Element& aEditingHost);
 
   /**
-   * HandleCSSIndentAtSelection() indents around Selection with CSS.
-   * NOTE: This is a helper method of `HandleIndentAtSelection()`.  If you
-   *       want to call this directly, you should check whether you need
-   *       do do something which `HandleIndentAtSelection()` does.
-   *
-   * @param aEditingHost        The editing host.
-   */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
-  HandleCSSIndentAtSelection(const Element& aEditingHost);
-
-  /**
    * HandleHTMLIndentAtSelection() indents around Selection with HTML.
    * NOTE: This is a helper method of `HandleIndentAtSelection()`.  If you
    *       want to call this directly, you should check whether you need
@@ -2369,6 +2355,7 @@
    *
    * @param aEditingHost        The editing host.
    */
+  // TODO: Make this take AutoRangeArray instead of retrieving `Selection`
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT EditActionResult
   HandleIndentAtSelection(const Element& aEditingHost);
 
